hudi-common/src/main/java/org/apache/hudi/common/model/HoodieFunctionalIndexDefinition.java

```diff
@@ -52,7 +55,7 @@ public HoodieFunctionalIndexDefinition(String indexName, String indexType, Strin
                                          Map<String, String> indexOptions) {
     this.indexName = indexName;
     this.indexType = indexType;
-    this.indexFunction = indexFunction;
+    this.indexFunction = nonEmpty(indexFunction) ? indexFunction : SPARK_IDENTITY;
```

Comment:
Can you elaborate the semantics for empty index function?

Reply:
if someone creates func index without specificying the function then we pick the identity function to be default.